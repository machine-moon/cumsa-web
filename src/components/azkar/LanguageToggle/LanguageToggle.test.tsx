import React from "react";
import { render } from "@testing-library/react";
import { LanguageToggle } from "./index";

describe("LanguageToggle", () => {
  const defaultProps = {
    mode: {
      arabic: true,
      transliteration: true,
      translation: true,
    },
    onChange: jest.fn(),
  };

  it("renders correctly with all options enabled", () => {
    const { container } = render(<LanguageToggle {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with only arabic enabled", () => {
    const { container } = render(
      <LanguageToggle
        {...defaultProps}
        mode={{
          arabic: true,
          transliteration: false,
          translation: false,
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with only translation enabled", () => {
    const { container } = render(
      <LanguageToggle
        {...defaultProps}
        mode={{
          arabic: false,
          transliteration: false,
          translation: true,
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with arabic and transliteration enabled", () => {
    const { container } = render(
      <LanguageToggle
        {...defaultProps}
        mode={{
          arabic: true,
          transliteration: true,
          translation: false,
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
