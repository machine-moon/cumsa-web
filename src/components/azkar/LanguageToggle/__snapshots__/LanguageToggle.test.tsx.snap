// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageToggle renders correctly with all options enabled 1`] = `
<div
  class="flex items-center gap-2 text-xs relative"
>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-[var(--blue)] border-[var(--blue)] text-white shadow-sm"
    type="button"
  >
    Arabic
  </button>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-[var(--blue)] border-[var(--blue)] text-white shadow-sm"
    type="button"
  >
    Transliteration
  </button>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-[var(--blue)] border-[var(--blue)] text-white shadow-sm"
    type="button"
  >
    Translation
  </button>
</div>
`;

exports[`LanguageToggle renders correctly with arabic and transliteration enabled 1`] = `
<div
  class="flex items-center gap-2 text-xs relative"
>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-[var(--blue)] border-[var(--blue)] text-white shadow-sm"
    type="button"
  >
    Arabic
  </button>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-[var(--blue)] border-[var(--blue)] text-white shadow-sm"
    type="button"
  >
    Transliteration
  </button>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-white border-[var(--blue)]/40 text-[var(--foreground)] hover:bg-[var(--blue)]/10"
    type="button"
  >
    Translation
  </button>
</div>
`;

exports[`LanguageToggle renders correctly with only arabic enabled 1`] = `
<div
  class="flex items-center gap-2 text-xs relative"
>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-[var(--blue)] border-[var(--blue)] text-white shadow-sm"
    type="button"
  >
    Arabic
  </button>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-white border-[var(--blue)]/40 text-[var(--foreground)] hover:bg-[var(--blue)]/10"
    type="button"
  >
    Transliteration
  </button>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-white border-[var(--blue)]/40 text-[var(--foreground)] hover:bg-[var(--blue)]/10"
    type="button"
  >
    Translation
  </button>
</div>
`;

exports[`LanguageToggle renders correctly with only translation enabled 1`] = `
<div
  class="flex items-center gap-2 text-xs relative"
>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-white border-[var(--blue)]/40 text-[var(--foreground)] hover:bg-[var(--blue)]/10"
    type="button"
  >
    Arabic
  </button>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-white border-[var(--blue)]/40 text-[var(--foreground)] hover:bg-[var(--blue)]/10"
    type="button"
  >
    Transliteration
  </button>
  <button
    class="relative px-2 py-1 rounded-md border text-xs transition-colors min-w-[88px] bg-[var(--blue)] border-[var(--blue)] text-white shadow-sm"
    type="button"
  >
    Translation
  </button>
</div>
`;
